W 12657814 restructuring preparingautomations se (#141)
mulesoft/docs-rpa: rpa-manager/modules/ROOT/pages/processautomation-prepare-project-application.adoc

= Manage Applications

> Process applications are third-party application installations that are required for the execution of a process. 

Reviewer: "to execute a process"?

Author: This surly can be worded better.
The content refers to a business process we want to automate. A business process consists of a series of steps a human executes to reach a certain goal.
For example:
1. Open an email with the application Microsoft Outlook.
2. Download and open the attachment.
3. Copy the account number, name, and amount to a data sheet with the application Microsoft Excel and save it.
To automate this business process the bot needs to access the applications Outlook and Excel.
In the Process Applications view we list all available applications and where to find them.
In the Bots view we assign those applications to bots that can access them.
In the Process Management view we assign the required application to the project.
Only bots with access to the 'required' applications can execute the process.

Author: How about:
Process applications are third-party application installations that business experts use to execute the process.
?

Reviewer: I like it!

Manage a list of all applications and their downtimes.

Create a new entry for each instance of an application, for example `FinWiz - Mythical University - Computer 1 - Tokyo` and `FinWiz - Mythical University - Computer 42 - San Francisco`.

> Applications are not available during down-times or while deactivated. 

Reviewer: ```suggestion
Applications are not available during downtimes or while deactivated. 
```

To avoid confusion, append the name of the time zone to the name of the application or the downtime. For example, name an application `Words - Tokyo` and a downtime `Electrical Power Outage Tokyo`.

In the *Process Management* view, a member of the project team xref:myrpa-start.adoc#required-process-applications[assigns required applications to a process] at process automation creation or phase transition.

In the *Bot Management* module, an administrator xref:botmanagement-manage.adoc#bot-assign-servicetimes-applications[assigns required applications to bots] that can access them. 

> A member of the process team can deploy a process on a bot that has assigned the required applications. If an application becomes non-available, the bot completes running processes using that application and restarts them when the downtime or deactivation period has ended.

Reviewer: ```suggestion
A member of the process team can deploy a process on a bot that has assigned the required applications. If an application becomes unavailable, the bot completes running processes using that application and restarts the processes when the downtime or deactivation period has ended.
```

== Before You Begin

> * Ask an administrator to xref:usermanagement-manage.adoc#assign-privileges-to-a-user[assign you the needed privileges]:

Reviewer: ```suggestion
* Ask an administrator to xref:usermanagement-manage.adoc#assign-privileges-to-a-user[assign you the required privileges]:
```

Reviewer: See previous comments on similar content.

+
[cols="1,2"]
|===
|*Action* |*Required Privileges*

|Open the *Process Automation* module
|Process Automation Open

|Create, change, and delete process applications
|Application Create, Application Edit, Application Delete

|===

> * For each application, note the timezone of the computer that it is installed on.

Reviewer: ```suggestion
* Note the timezone of the computer that each application is installed on.
```

== View All Applications

To view all existing applications for automation projects, open the *Process Applications* view of the *Process Automation* module.

== Create a Process Application

Create one entry for each application instance required for the execution of processes.

To create a process application:

. Open the *Process Applications* view of the *Process Automation* module.
. Click *Create*.
. [[form-create-process-applications]] Complete the *Create a new application* form:
+
* *Time Zone*:
+
Time zone for defining downtimes.
+

> Specify a downtime in the time zone in which it occurs. If the bot that you assign the downtime to is installed on a computer that uses a different time zone, the downtime interval is converted to the bot’s time zone. If you assign the application to multiple bots in different time zones, all bots pause executing processes that use that application simultaneously. The time intervals might change, depending on daylight saving time.

Reviewer: What does "it" refer to here?

Author: to downtime

Author: Is that ambiguous?  A time zone cannot occur.

Reviewer: I think it's fine. Ignore previous comment :)

+
For example, if you specify a downtime that starts at 1:30 AM and ends at 5:30 AM in Tokyo time (JST (UTC +9)) and the bot is installed on a computer in San Francisco (PST (UTC -8)), the bot does not execute processes between 8:30 AM and 12:30 PM in San Francisco time.
+
To avoid confusion, append the name of the time zone to the name of the downtime. For example, name your downtime `Electrical Power Outage Tokyo`.
+
* *Downtimes*
+
Click *Create* to add a downtime to the application.
. Click *Save*.

== Edit an Application

> Edit an application to change its data or down-times:

Reviewer: ```suggestion
Edit an application to change its data or downtimes:
```

Reviewer: global change: downtime

Author: You do not need to repeat yourself. I always change every occurrence of misspelled words :) But thank you nevertheless for suggesting a global change.

Reviewer: Noted! Some other people only make the requested changes :)

. Open the *Process Applications* view of the *Process Automation* module.
. Click *Create*.

> . Edit the *Edit the application* form.

Reviewer: Edit > Complete?

Reviewer: No. Leave as is.

+
For an explanation of the properties, see  <<form-create-process-applications, *Create an Application*>>.
. Click *Save*.

== Deactivate or Activate a Process Application

> Deactivate an application if it is temporarily not available and you don't know when it will be available again.

Reviewer: ```suggestion
Deactivate an application if it is temporarily unavailable and you don't know when it will be available again.
```

> Use down-times to schedule maintenance and planned downtimes. Deactivation periods are not added to the list of down-times.

Reviewer: Is down-times in this context part of the UI?

Author: You can make the application unavailable via the UI in two ways:
- Inside the Create/Edit Application form (you need to click a button / an icon to open the form a view and add to  the downtimes list)
- By clicking the Deactivate icon on the panel of the application in the Applications view 
The latter is described in this paragraph.

Should I add a link to the downtimes property in the creation form?

Reviewer: I think I was wondering about the two different spellings in the same sentence - it would make sense to have the hyphenated version if it appeared in the UI.

> Applications are not available during down-times or while deactivated. A member of the process team can deploy a process on a bot that has assigned the required processes. If an application becomes non-available, the bot completes running processes using that application and restarts them when the downtime or deactivation period has ended.

Reviewer: ```suggestion
Applications are not available during down-times or while deactivated. A member of the process team can deploy a process on a bot that has assigned the required processes. If an application becomes unavailable, the bot completes running processes using that application and restarts them when the downtime or deactivation period has ended.
```

If an application is not available, deactivate it:

. Open the *Process Applications* view of the *Process Automation* module.
. Click the *Deactivate* icon image:deactivate-icon.png[toggle-on symbol,1.5%,1.5%] on the panel of the application you want to deactivate.

RPA Manager displays an *Inactive* label beneath the logo of the application.

If a deactivated application becomes available again, activate it:

. Open the *Process Applications* view of the *Process Automation* module.

> . Click the *Activate* icon image:activate-icon.png[toggle-off symbol,1.5%,1.5%] on the panel of the application you want to activate.

Reviewer: Global: remove "you want"

Author: Removed all that I easily could. Added [W-12942071](https://gus.lightning.force.com/a07EE00001OLu4uYAD) to the backlog for checking the rest...

Reviewer: No worries. It's a niggly thing, not an error per se.

> RPA Manager displays an *Active* label beneath the logo of the application.

Reviewer: I think beneath might be inaccessible (like below).

Author: replaced with 'on the panel of the application'.

== Remove a Process Application

Remove an application if it is no longer available:

. Open the *Process Applications* view of the *Process Automation* module.
. Click the the *Remove* icon image:delete-icon.png[trash symbol,1.5%,1.5%] on the panel of the application you want to remove.
. Confirm the removal.

> RPA Manager removes the application from all processes and bots to which it was assigned to.

Reviewer: ```suggestion
RPA Manager removes the application from all processes and bots to which it was assigned.
```
